diff --git a/nixos/modules/misc/nixpkgs.nix b/nixos/modules/misc/nixpkgs.nix
index bfcae9c7a935..cc5f57421092 100644
--- a/nixos/modules/misc/nixpkgs.nix
+++ b/nixos/modules/misc/nixpkgs.nix
@@ -379,16 +379,6 @@ in
           the legacy definitions.
         '';
       }
-      {
-        assertion = opt.pkgs.isDefined -> cfg.config == {};
-        message = ''
-          Your system configures nixpkgs with an externally created instance.
-          `nixpkgs.config` options should be passed when creating the instance instead.
-
-          Current value:
-          ${lib.generators.toPretty { multiline = true; } opt.config}
-        '';
-      }
     ];
   };
 
